### DISCUSSION TITLE
Who should set move labels on BareMetalHost CRD during pivoting

### DISCUSSION THREAD

**Discussion by fmuyassarov** [metal3-io/metal3-dev-env#1007] [Q&A]
CAPM3 provider handles some external objects like BareMetalHost, HardwareData during pivoting. Since those objects are not tight to a Cluster object by default, we set `clusterctl.cluster.x-k8s.io` and `cluster.x-k8s.io/cluster-name` labels so that clusterctl moves them too. Currently, CAPM3 controller sets only `cluster.x-k8s.io/cluster-name` label, while `clusterctl.cluster.x-k8s.io` is left out to a user to set.

I believe it is better to have a single entity setting those labels, whether it be a controller that would do it automatically or a user. Do you think we should leave it to a user to set it or enforce the controller to do it?

**Comment by fmuyassarov** [comment]
To me it seems that controller setting those labels is correct thing. Because, pivot operation should work out of the box. One example, CAPI sets some of those labels too on the its objects instead of leaving it to a user. I understand that CAPM3 is unique provider since we have external component BMO, and we still need to handle BMO and Ironic installation on the target cluster manually. But even in that case, setting labels is something that can be handled by the controller, which will make user's life easier and boring.

**Reply by lentzi90** [reply]
+1 from me. It would be nice if the controller sets those labels.
